stdlib/public/stubs/CMakeLists.txt

```diff
@@ -18,6 +18,18 @@ set(LLVM_OPTIONAL_SOURCES
     ${swift_stubs_objc_sources}
     ${swift_stubs_unicode_normalization_sources})
 
+# ICU isn't required on Darwin, but is on every other platform.
+# Now in case we're cross-compiling from Darwin for another platform,
+# the find_package should still be executed.
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(icu_required, "REQUIRED")
+else()
+  set(icu_required, "")
+endif()
+
+find_package(ICU ${icu_required} COMPONENTS uc)
+set(ICU_UC_LIBRARY "")
```

Comment:
For some reason I was totally looking over the problem pointed to by @jrose-apple .. It took @llvm-beanz 's comments to make me realise the problematic REQUIRED logic..
I was under the impression that the whole find_package() function call was deemed inappropriate, hence my comments and misunderstanding.
@jrose-apple  , @llvm-beanz  : can you please look at commit 6c355d74138c1adeebb6d7ed33e38601b2a79160 ? It fixes the problematic logic.